distributed/diagnostics/nvml.py

```diff
@@ -134,29 +134,64 @@ def _pynvml_handles():
     count = device_get_count()
     if NVML_STATE == NVMLState.DISABLED_PYNVML_NOT_AVAILABLE:
         raise RuntimeError("NVML monitoring requires PyNVML and NVML to be installed")
-    elif NVML_STATE == NVMLState.DISABLED_LIBRARY_NOT_FOUND:
+    if NVML_STATE == NVMLState.DISABLED_LIBRARY_NOT_FOUND:
         raise RuntimeError("PyNVML is installed, but NVML is not")
-    elif NVML_STATE == NVMLState.DISABLED_WSL_INSUFFICIENT_DRIVER:
+    if NVML_STATE == NVMLState.DISABLED_WSL_INSUFFICIENT_DRIVER:
         raise RuntimeError(
             "Outdated NVIDIA drivers for WSL, please upgrade to "
             f"{MINIMUM_WSL_VERSION} or newer"
         )
-    elif NVML_STATE == NVMLState.DISABLED_CONFIG:
+    if NVML_STATE == NVMLState.DISABLED_CONFIG:
         raise RuntimeError(
             "PyNVML monitoring disabled by 'distributed.diagnostics.nvml' "
             "config setting"
         )
-    elif count == 0:
+    if count == 0:
         raise RuntimeError("No GPUs available")
-    else:
-        try:
-            gpu_idx = next(
-                map(int, os.environ.get("CUDA_VISIBLE_DEVICES", "").split(","))
+
+    device = 0
+    cuda_visible_devices = os.environ.get("CUDA_VISIBLE_DEVICES", "")
+    if cuda_visible_devices:
+        device = _parse_cuda_visible_device(cuda_visible_devices.split(",")[0])
+    return _get_handle(device)
+
+
+# Port from https://github.com/rapidsai/dask-cuda/blob/0f34116c4f3cdf5dfc0df0dbfeba92655f686716/dask_cuda/utils.py#L403-L437
+def _parse_cuda_visible_device(dev):
```

Comment:
Would it make sense to relocate that function? If so, what's the best place?

Reply:
I hate that we have a lot of code duplication between Dask-CUDA and Distributed, ideally none of this "low-level" PyNVML error handling should exist in either Dask-CUDA nor Distributed. I have a better solution ready for all that but unfortunately I hit some barriers on distribution issues internally. How pressing is this work? Is this something that could wait, say another month, or does this need to go in ASAP?